test/clj/harja/palvelin/komponentit/todennus_test.clj

```diff
@@ -85,25 +85,45 @@
                                               13343 #{"ELY_Urakanvalvoja"}}}]
     (is (= vastaus odotetut-roolit))))
 
-(defn testi-cognito-headerit []
-  (let [x-iam-data [{"typ" "JWT"
-                     "kid" "7d2ed764-76dd-44c3-b4cf-8cde89fe6e5f"
-                     "alg" "ES256"
-                     "iss" "https://cognito-idp.eu-west-1.amazonaws.com/foobar"
-                     "client" "3ctc20d3i4ghv34ks0semt4e16"
-                     "signer" "arn:aws:elasticloadbalancing:eu-west-1:083539282917:loadbalancer/app/foobar/8dad8bb767eb8568"
-                     "exp" 1687175356}
-                    {"custom:rooli" "2234567-8_Paakayttaja"
-                     "custom:sukunimi" "Destialainen"
-                     "custom:ytunnus" "2163026-3"
-                     "email" "daniel@example.com"
-                     "exp" 1687175356
-                     "custom:uid" "daniel"
-                     "custom:puhelin" "1234567890"
-                     "custom:organisaatio" "Destia Oy"
-                     "custom:etunimi" "Daniel"}
-                    ;; Tämä on vain signature. Ei relevantti näiden testien kannalta tällä hetkellä.
-                    "TDZJ0uQA-H2GEfw38cVc-OS8gAsRVlW_EyPojJOtLKbqMalXUcq59BFB-ZJY1UXmxhdNDX04IEAQs70qa5p2Gw=="]
+(def test-entraid-json-headerit
+  [{"typ" "JWT"
+    "kid" "7d2ed764-76dd-44c3-b4cf-8cde89fe6e5f"
+    "alg" "ES256"
+    "iss" "https://cognito-idp.eu-west-1.amazonaws.com/foobar"
+    "client" "3ctc20d3i4ghv34ks0semt4e16"
+    "signer" "arn:aws:elasticloadbalancing:eu-west-1:083539282917:loadbalancer/app/foobar/8dad8bb767eb8568"
+    "exp" 1687175356}
+   {"custom:rooli" "[\"Jarjestelmavastaava\",\"MHU-TESTI-LAP-ROV_vastuuhenkilo\",\"MHU-TESTI-LAP-IVA_vastuuhenkilo\"]"
```

Comment:
Tässä vois käyttää:

```clojure
(json/write-str ["Jarjestelmavastaava" "MHU-TESTI-LAP-ROV_vastuuhenkilo" "MHU-TESTI-LAP-IVA_vastuuhenkilo"])
```

Oisko parempi näin? Molemmilla sama lopputulos kyllä.

Reply:
Kyllähän se fiksumpaa on, kuin alkaa itse escapettamaan manuaalisesti noita

Reply:
Laitoin